### PR TITLE
Fixed minor bugs in sto

### PR DESCRIPTION
**Pull request by maxsam4**
### Internal
- Minimum investment amount is now checked correctly against the payment tokens instead of the offering token.
- Accounted for `price_divisor` when checking for max price.
- No longer checks that the sender is an asset holder since 0 amount transfers are allowed.

### CODE REVIEW

**Review by CJP10 — COMMENTED** [review]
LGTM so far

File: pallets/sto/src/lib.rs [hunk 1]
```diff
@@ -378,8 +377,9 @@ decl_module! {
             }
 
             ensure!(remaining == 0.into(), Error::<T>::InsufficientTokensRemaining);
+            ensure!(cost >= fundraiser.minimum_investment, Error::<T>::InvestmentAmountTooLow);
             ensure!(
-                max_price.map(|max_price| cost <= max_price * investment_amount).unwrap_or(true),
+                max_price.map(|max_price| cost <= max_price.saturating_mul(purchase_amount) / price_divisor).unwrap_or(true),
```
Comment (CJP10) [inline pallets/sto/src/lib.rs:382]: Should we check that max_price isn't Some(0) to avoid div by zero?

Comment (maxsam4) [reply pallets/sto/src/lib.rs:382]: `max_price` is in the numerator, not the denominator so I don't think this is an issue.

**Review by CJP10 — APPROVED** [review]

**Review by Centril — CHANGES_REQUESTED** [review]

File: pallets/sto/src/lib.rs (hunk 1 above)
Comment (fmiguelgarcia) [reply pallets/sto/src/lib.rs:384]: Not related with this changes but It looks like `price_divisor` should be a common constant.

**Review by fmiguelgarcia — APPROVED** [review]

**Review by adamdossa — APPROVED** [review]

**Review by Centril — APPROVED** [review]

**Pull request merged**